src/tema4/Main.java

```diff
@@ -0,0 +1,360 @@
+            animalx.setId(createVariableString("A"));
+            System.out.println("Please provide the storage temperature in Celsius");
+            animalx.setStorageTemperature(Float.parseFloat(createVariableString("float")));
+            System.out.println("Please provide the quantity for the product");
+            animalx.setStock(Integer.parseInt(createVariableString("intPositive")));
+            System.out.println("Please provide the weight of the product");
+            animalx.setWeight(Float.parseFloat(createVariableString("float")));
+            System.out.println("Please provide the price per product");
+            animalx.setPrice(Float.parseFloat(createVariableString("float")));
+            System.out.println("Please provide the expiry date using the format YYYY-MM-DD");
+            animalx.setValidityDate(createVariableString("date"));
+            listOfAnimalProducts.add(animalx);
+
+        }else{
+            Vegetable animalx = new Vegetable();
+            System.out.println("Please provide the name of the vegetable product");
+            animalx.setId(createVariableString("A"));
+            System.out.println("Please provide the list of vitamins for the product");
+            animalx.setListOfVitamin(createVariableString("a"));
+            System.out.println("Please provide the quantity for the product");
+            animalx.setStock(Integer.parseInt(createVariableString("intPositive")));
+            System.out.println("Please provide the weight of the product");
+            animalx.setWeight(Float.parseFloat(createVariableString("float")));
+            System.out.println("Please provide the price per product");
+            animalx.setPrice(Float.parseFloat(createVariableString("float")));
+            System.out.println("Please provide the expiry date using the format YYYY-MM-DD");
+            animalx.setValidityDate(createVariableString("date"));
+            listOfVegetablesProducts.add(animalx);
+            for (Vegetable x : listOfVegetablesProducts){
+                System.out.println(x.getId() + x.getListOfVitamin());
+            }
+        }
+    }
+
+    /**
+     * create a string with the input received from user and checks if the string value may be converted into a certain type based on the param
+     * for int will check if int
+     * for float will check if float
+     * for intPositive will check if integer positive
+     * date will check if date is valid and bigger then today
+     * dateSell will check if valid and is not bigger then today
+     * anything else will just return the String
+     * if the input received from user does not meet the checks requested by the param a new value will be requested
+     * @param a may have any value but for int, float, intPositive, date, dateSell will do checks
+     * @return a string value that meets the requirements in the param
+     */
+    private static String createVariableString(String a) {
```

Comment:
while this method does provide you with the retry option of inserting a value. It is a very complicated way of doing it.
For the level you are at right now, I would drop this as it overly complicated things and not in a good way.

As an explanation:
- you are using String "int", "float", etc to signal what you want to check. You should either use constants for this or enums
- you are doing readNumber.nextLine() --- sooo you're reading a String using a variable that's called readNumber, but you also read dates?... maybe scanner would be a better name for that variable.
- you are reading a String, you do a switch case on another String, to parse the String that was read to a certain type, and if that succeeds you return a String so that the calling method can parse that String again... (you see why I'm saying this is a very complicated way of doing it)
- the name "createVariableString" is confusing... you actually want to read from the keyboard a specific number/date... why are you creating a variable String?

Since this is a simple example. A better way would have been to create a method for each field you want to read, and do the propper validations there.
private Date readSellDate()... etc
It would have looked cleaner